### ISSUE TITLE
Memory leak in watch mode

### ISSUE THREAD

**Issue by lorenzodallavecchia** [OpenNTF/dojo-webpack-plugin#168]
When using webpack in watch mode (or with `webpack-dev-server`) the plugin causes a massive memory leak.

I think the offending code is at least in `DojoLoaderPlugin.js` and `DojoAMDModuleFactoryPlugin.js`.

```javascript
tap(compiler, "compilation", (compilation, params) => {
	const context = Object.create(this, {
		compilation:{value: compilation},
		params:{value: params}
	});

	// later (for example)
	tap(compiler, "make", this.validateEmbeddedLoader, context);
});
```

This causes each `compilation` object to be retained by the `compiler` object (to be used as `this` for the hook). Since each compilation contains compiled assets, this may leak several MBs after each incremental compilation run.

Another side effect is that the old hooks stay around and are executed, gradually slowing down webpack and possibly causing unintended behavior.

I don't know how to fix this quickly. We should either refactor the plugin to not rely on the state being carried along the hook function, or we should find a clever hack to "unregister" the stale hooks, which is something not allowed by `tapable` out of the box.

**Comment by chuckdumont** [comment]
Should be much better now.  Please try out version 2.7.3-5

**Comment by lorenzodallavecchia** [comment]
I just tried with 2.7.3-5 and it appears to be fixed: no more `Compilation` objects leaking. 
Thanks @chuckdumont!

**Comment by chuckdumont** [comment]
Thanks.  I just published 2.7.3
